src/contracts/exchange/NiftyswapExchange20.sol

```diff
@@ -455,6 +455,60 @@ contract NiftyswapExchange20 is ReentrancyGuard, ERC1155MintBurn, INiftyswapExch
     emit LiquidityAdded(_provider, _tokenIds, _tokenAmounts, currencyAmounts);
   }
 
+  /**
+   * @dev Convert pool participation into amounts of token and currency.
+   * @dev Rounding error of the asset with lower resolution is traded for the other asset.
+   * @param _amountPool       Participation to be converted to tokens and currency.
+   * @param _tokenReserve     Amount of tokens on the AMM reserve.
+   * @param _currencyReserve  Amount of currency on the AMM reserve.
+   * @param _totalLiquidity   Total liquidity on the pool.
+   *
+   * @return currencyAmount Currency corresponding to pool amount plus rounded tokens.
+   * @return tokenAmount    Token corresponding to pool amount plus rounded currency.
+   */
+    function _toRoundedLiquidity(
+    uint256 _amountPool,
+    uint256 _tokenReserve,
+    uint256 _currencyReserve,
+    uint256 _totalLiquidity
+  ) internal pure returns (
+    uint256 currencyAmount,
+    uint256 tokenAmount,
+    int256  tradedCurrency,
+    int256  tradedToken
+  ) {
+    uint256 currencyProduct = _amountPool.mul(_currencyReserve);
+    uint256 tokenProduct = _amountPool.mul(_tokenReserve);
+
+    if (tokenProduct > currencyProduct) {
+      // Convert all currencyProduct rest to token
+      uint256 currencyRest = currencyProduct % _totalLiquidity;
+      uint256 boughtToken = getSellPrice(currencyRest, _currencyReserve, _tokenReserve);
+
+      // This is not needed, the division removes the rest
+      // currencyProduct -= currencyRest;
+      tokenProduct += boughtToken;
+
+      tradedCurrency = -int256(currencyRest);
```

Comment:
Why negative? Not sure I understand sorry 😅

Reply:
This is just used on the event, instead of having two distinct events selling and buying, it uses the same one but the direction of the trade is defined by the sign of the amount.